perception/traffic_light_classifier/src/single_image_inference_node.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/rclcpp.hpp>
+
+#if ENABLE_GPU
+#include <traffic_light_classifier/cnn_classifier.hpp>
+#endif
+
+#include <traffic_light_classifier/color_classifier.hpp>
+#include <traffic_light_classifier/nodelet.hpp>
+
+#include <memory>
+#include <string>
+
+namespace
+{
+std::string toString(const uint8_t state)
+{
+  if (state == autoware_auto_perception_msgs::msg::TrafficLight::RED) {
+    return "red";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::AMBER) {
+    return "yellow";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::GREEN) {
+    return "green";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::WHITE) {
+    return "white";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::CIRCLE) {
+    return "circle";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::LEFT_ARROW) {
+    return "left";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::RIGHT_ARROW) {
+    return "right";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::UP_ARROW) {
+    return "straight";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_ARROW) {
+    return "down";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_LEFT_ARROW) {
+    return "down_left";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_RIGHT_ARROW) {
+    return "down_right";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::CROSS) {
+    return "cross";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::UNKNOWN) {
+    return "unknown";
+  } else {
+    return "";
+  }
+}
+}  // namespace
+
+namespace traffic_light
+{
+class SingleImageInferenceNode : public rclcpp::Node
```

Comment:
What is difference between this node and TrafficLightClassifierNodelet?

Reply:
This node is for debug?

Reply:
Yes, this is just a debug node to test inference with a image file.

Reply:
Can you please write the word `debug` into the file or directory name and class name so that we can understand for debugging?

Reply:
I addressed [here](https://github.com/autowarefoundation/autoware.universe/pull/2820/commits/b748a82bee5eb152a29dbb78305e94fb4e2b2b42)